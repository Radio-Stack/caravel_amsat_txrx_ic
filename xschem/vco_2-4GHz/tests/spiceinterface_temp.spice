**.subckt vco_2-4GHz_tb
V1 vctl 0 vctl
Vdd net1 0 vdd
xvco net1 Ip In Qp Qn NCp NCn FBp FBn vctl 0 vco_2-4GHz
**** begin user architecture code


.param temp=125.000000
.temp 125.000000

.save v(ip) v(in) v(vctl) i(vdd)

.include sky130_fd_pr/cells/rf_pfet_01v8_lvt/sky130_fd_pr__rf_pfet_01v8_lvt_aM04W5p00L0p35.spice
.lib sky130_fd_pr/models/sky130.lib.spice lh
.tran 0.01n 10n uic
.param vdd=1.800000
.param vctl=1.800000


**** end user architecture code
**.ends

* expanding   symbol:  vco_2-4GHz/vco_2-4GHz.sym # of pins=11

.subckt vco_2-4GHz  VDD ph1_p ph1_n ph2_p ph2_n ph3_p ph3_n ph4_p ph4_n CTL GND
*.ipin CTL
*.iopin VDD
*.iopin GND
*.opin ph3_p
*.opin ph3_n
*.opin ph4_p
*.opin ph4_n
*.opin ph2_p
*.opin ph2_n
*.opin ph1_p
*.opin ph1_n
x1 VDD net2 net1 net7 net8 GND CTL vco_2-4Ghz_delaycell
x2 VDD net4 net3 net2 net1 GND CTL vco_2-4Ghz_delaycell
x3 VDD net6 net5 net4 net3 GND CTL vco_2-4Ghz_delaycell
x4 VDD net8 net7 net6 net5 GND CTL vco_2-4Ghz_delaycell
x5 VDD ph1_p ph1_n net1 net2 GND rf_bufferdiff
x6 VDD ph2_p ph2_n net3 net4 GND rf_bufferdiff
x7 VDD ph3_p ph3_n net5 net6 GND rf_bufferdiff
x8 VDD ph4_p ph4_n net7 net8 GND rf_bufferdiff
I0 GND GND pulse(1000n 0 0 400p 100p 100p 10 20 0)
.ends


* expanding   symbol:  vco_2-4GHz_delaycell/vco_2-4Ghz_delaycell.sym # of pins=7

.subckt vco_2-4Ghz_delaycell  VDD OUTn OUTp Pn Pp GND CTL
*.ipin Pp
*.ipin Pn
*.opin OUTn
*.opin OUTp
*.iopin VDD
*.iopin GND
*.ipin CTL
XXMprimpos OUTp Pp GND GND sky130_fd_pr__rf_nfet_01v8_bM04W5p00L0p15 m=3
XXMcrossneg OUTn OUTp GND GND sky130_fd_pr__rf_nfet_01v8_bM02W3p00L0p15 m=1
XXMctlneg OUTn CTL VDD VDD sky130_fd_pr__rf_pfet_01v8_lvt_aM04W5p00L0p35 m=1
XXMctlpos OUTp CTL VDD VDD sky130_fd_pr__rf_pfet_01v8_lvt_aM04W5p00L0p35 m=1
XXMcrosspos OUTp OUTn GND GND sky130_fd_pr__rf_nfet_01v8_bM02W3p00L0p15 m=1
XXMprimneg OUTn Pn GND GND sky130_fd_pr__rf_nfet_01v8_bM04W5p00L0p15 m=3
XXMctlpos1 net2 GND VDD VDD sky130_fd_pr__rf_pfet_01v8_bM02W5p00L0p15 m=1
XXMctlneg1 net1 GND VDD VDD sky130_fd_pr__rf_pfet_01v8_bM02W5p00L0p15 m=1
XXMctlpos2 OUTp GND net2 VDD sky130_fd_pr__rf_pfet_01v8_bM02W5p00L0p15 m=1
XXMctlneg2 OUTn GND net1 VDD sky130_fd_pr__rf_pfet_01v8_bM02W5p00L0p15 m=1
.ends


* expanding   symbol:  rf_bufferdiff/rf_bufferdiff.sym # of pins=6

.subckt rf_bufferdiff  vdd out_n out_p in_p in_n vss
*.iopin vdd
*.iopin vss
*.ipin in_p
*.ipin in_n
*.opin out_n
*.opin out_p
XXMdiffp out_n in_p vss vss sky130_fd_pr__rf_nfet_01v8_bM02W1p65L0p15 m=1
XXMdiffn out_p in_n vss vss sky130_fd_pr__rf_nfet_01v8_bM02W1p65L0p15 m=1
XRn vdd out_n vss sky130_fd_pr__res_xhigh_po_2p85 m=1
XRp vdd out_p vss sky130_fd_pr__res_xhigh_po_2p85 m=1
.ends

.end
