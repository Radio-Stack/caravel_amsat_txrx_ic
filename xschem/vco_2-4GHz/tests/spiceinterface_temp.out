codemodel: no such command available in ngspice
codemodel: no such command available in ngspice
codemodel: no such command available in ngspice
codemodel: no such command available in ngspice
codemodel: no such command available in ngspice
codemodel: no such command available in ngspice

Circuit: **.subckt vco_2-4ghz_tb

Scale set
Doing analysis at TEMP = 125.000000 and TNOM = 27.000000

Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: Source conductance reset to 1.0e3 mho.
Warning: include: has no value, DC 0 assumed
Warning: i.xvco.i0: no DC value, transient time 0 value used
No. of Data Columns : 5  

Initial Transient Solution
--------------------------

Node                                   Voltage
----                                   -------
vctl                                         0
net1                                         0
xvco.net1                                    0
xvco.net8                                    0
xvco.net2                                    0
xvco.net7                                    0
xvco.x1.net2                                 0
xvco.x1.net1                                 0
xvco.net3                                    0
xvco.net4                                    0
xvco.x2.net2                                 0
xvco.x2.net1                                 0
xvco.net5                                    0
xvco.net6                                    0
xvco.x3.net2                                 0
xvco.x3.net1                                 0
xvco.x4.net2                                 0
xvco.x4.net1                                 0
ip                                           0
in                                           0
xvco.x5.xrn.xsky130_fd_pr__res_xhigh_po_2p85.ra               0
xvco.x5.xrp.xsky130_fd_pr__res_xhigh_po_2p85.ra               0
qp                                           0
qn                                           0
xvco.x6.xrn.xsky130_fd_pr__res_xhigh_po_2p85.ra               0
xvco.x6.xrp.xsky130_fd_pr__res_xhigh_po_2p85.ra               0
ncp                                          0
ncn                                          0
xvco.x7.xrn.xsky130_fd_pr__res_xhigh_po_2p85.ra               0
xvco.x7.xrp.xsky130_fd_pr__res_xhigh_po_2p85.ra               0
fbp                                          0
fbn                                          0
xvco.x8.xrn.xsky130_fd_pr__res_xhigh_po_2p85.ra               0
xvco.x8.xrp.xsky130_fd_pr__res_xhigh_po_2p85.ra               0
sky130_fd_pr__esd_nfet_05v0_nvt.pm3               0
                                             0
vdd#branch                                   0
v1#branch                                    0


No. of Data Rows : 1026

Total analysis time (seconds) = 1.71965

Total CPU time (seconds) = 51.261 

Total DRAM available = 7908.293 MB.
DRAM currently available =  153.211 MB.
Total ngspice program size =  243.086 MB.
Resident set size =  223.559 MB.
Shared ngspice pages =    6.891 MB.
Text (code) pages =    5.105 MB.
Stack = 0 bytes.
Library pages =  217.039 MB.

